### PR TITLE
Android 11507 migrate high lighted card compose

### PR DESCRIPTION
**Pull request by afernandezalabarce**
[Task](https://jira.tid.es/browse/ANDROID-11507)

### :goal_net: What's the goal?
Repair some bugs and integrate HighLightedCard into mistica catalog

### :construction: How do we do it?
Repair some bugs and integrate with Mistica catalog
Compose catalog for mistica are developed with reactive ui changes.

### ☑️ Checks
- [X] I updated the documentation, including readmes and wikis. If this is a breaking change, update [UPGRADING.md](../UPGRADING.md) to inform users how to proceed. If no updates are necessary, indicate so.
- [X] Tested with dark mode.
- [X] Tested with API 23.

### :test_tube: How can I test this?
_If it cannot be tested explain why._
- [ ] 🖼️ Screenshots/Videos
- [ ] Mistica App QR or download link
- [ ] Reviewed by Mistica design team
- [ ] ...

### CODE REVIEW

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt [hunk 1]
```diff
@@ -0,0 +1,269 @@
+package com.telefonica.mistica.catalog.ui.compose.components
```
Comment (afernandezalabarce) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt:1]: this file is the reactive ui for compose with the goal of check ui of HighLightedCard composable

File: library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt [hunk 2]
```diff
@@ -50,7 +64,15 @@ fun HighLightedCard(
         border = BorderStroke(width = 1.dp, color = MisticaTheme.colors.border)
     ) {
         ConstraintLayout(
-            modifier = modifier.padding(bottom = if(button.buttonStyle != null) 0.dp else 16.dp)
+            modifier = modifier.drawBehind {
```
Comment (afernandezalabarce) [inline library/src/main/java/com/telefonica/mistica/compose/card/highlightedcard/HighLightedCard.kt:67]: to paint background, due lacks on paint with old format, are best to paint with drawbehind.

**Comment by gmerinojimenez** [comment]
Compose:
image
XML:
image

The colors doesn't match between implementations, can you check it? :)

**Comment by gmerinojimenez** [comment]
There should be a margin between the text and the image:
XML:
image
Compose:
image

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                }
+                .padding(16.dp)
+                .verticalScroll(
+                    state = scrollState,
+                    enabled = true,
+                )
+        ) {
+            Text(text = "HIGHLIGHTED CARD TESTER")
+
+            OutlinedTextField(
```
[the remark this answers is not in the document]
Comment (afernandezalabarce) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt]: Done!

**Review by haynlo — APPROVED** [review]
Good job

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt
Comment (pmartinbTEF) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt:1]: In the classic tab of the highlighted card when you press any card button (close, primary button, ...) it shows a Toast indicating that it has been pressed. Maybe you should add this behaviour in compose tab too

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+                Text(text = "None")
+            }
+
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .clickable { onButtonTypeSelected(buttonConfig.copy(buttonStyle = ButtonStyle.PRIMARY)) },
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                RadioButton(selected = buttonConfig.buttonStyle == ButtonStyle.PRIMARY, onClick = { onButtonTypeSelected(buttonConfig.copy(buttonStyle = ButtonStyle.PRIMARY)) })
```
Comment (pmartinbTEF) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt:239]: Line too long

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+                Text(text = "Primary")
+            }
+
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .clickable { onButtonTypeSelected(buttonConfig.copy(buttonStyle = ButtonStyle.SECONDARY)) },
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                RadioButton(selected = buttonConfig.buttonStyle == ButtonStyle.SECONDARY, onClick = { onButtonTypeSelected(buttonConfig.copy(buttonStyle = ButtonStyle.SECONDARY)) })
```
Comment (pmartinbTEF) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt:249]: Line too long

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+                Text(text = "Secondary")
+            }
+
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .clickable { onButtonTypeSelected(buttonConfig.copy(buttonStyle = ButtonStyle.LINK)) },
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                RadioButton(selected = buttonConfig.buttonStyle == ButtonStyle.LINK, onClick = { onButtonTypeSelected(buttonConfig.copy(buttonStyle = ButtonStyle.LINK)) })
```
Comment (pmartinbTEF) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt:259]: Line too long

**Comment by pmartinbTEF** [comment]
Different bottom margin for the fit image.
Classic:
Captura de pantalla de 2023-05-10 13-14-26
Compose:
Captura de pantalla de 2023-05-10 13-14-38

**Review by pmartinbTEF — CHANGES_REQUESTED** [review]
Good job! please check comments

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt (hunk 5 above)
Comment (afernandezalabarce) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt]: done!

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt (hunk 6 above)
Comment (afernandezalabarce) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/HighLightedCards.kt]: done!

**Review by pmartinbTEF — APPROVED** [review]
Great!

**Review by gmerinojimenez — APPROVED** [review]
Good job! thank you!

**Pull request merged**